src/js-display-panel.ts

```diff
@@ -0,0 +1,135 @@
+    const hidePanelButton = document.createElement('button');
+    hidePanelButton.classList.add('display-panel__hide');
+    hidePanelButton.innerHTML = 'Hide Panel';
+
+    return hidePanelButton;
+  }
+
+  private createToggleList(): HTMLUListElement {
+    return document.createElement('ul');
+  }
+
+  private createToggleListItem(toggle: ToggleStatus): HTMLLIElement {
+    const li = document.createElement('li');
+    li.appendChild(this.createToggleStatusNode(toggle));
+    li.appendChild(this.createToggleNameNode(toggle.Name));
+
+    return li;
+  }
+
+  private createToggleNameNode(name: string): Element {
+    const nameNode = document.createElement('div');
+    nameNode.classList.add('item-name');
+    nameNode.innerHTML = name;
+
+    return nameNode;
+  }
+
+  private createToggleStatusNode(toggle: ToggleStatus): Element {
+    const featureToggleUpdateLink = this.createFeatureToggleLink(toggle);
+    
+    const statusNode = document.createElement('label');
+    statusNode.classList.add('item-status');
+    const toggleCheckbox = document.createElement('input');
+    toggleCheckbox.setAttribute('type', 'checkbox');
+    
+    const sliderDisplayNode = document.createElement('span');
+    sliderDisplayNode.classList.add('slider');
+    sliderDisplayNode.setAttribute('data-reload-href', featureToggleUpdateLink);
+    sliderDisplayNode.addEventListener('click', e => {
+      e.preventDefault();
+
+      this.reloadPageWithUpdatedQueryString(featureToggleUpdateLink);
+    });
+
+    statusNode.appendChild(toggleCheckbox);
+    statusNode.appendChild(sliderDisplayNode);
+
+    if (toggle.IsActive)
+      toggleCheckbox.setAttribute('checked', 'true');
+    else
+      toggleCheckbox.removeAttribute('checked');
+
+    return statusNode;
+  }
+
+  private createFeatureToggleLink(toggle: ToggleStatus): string {
+    const featureOnOffString = toggle.IsActive ? 'featureoff' : 'featureon',
+          queryString = this.getQueryString(`${featureOnOffString}=${toggle.Name}`);
+    return `${window.location.origin}${window.location.pathname}?${queryString}${window.location.hash}`;
+  }
+
+  private reloadPageWithUpdatedQueryString(updateLink: string): void {
+    setTimeout(() => {
+      window.location.href = updateLink;
+    }, 200);
+  }
+
+  private getQueryString(featureCommandQueryString: string): string {
+    const qsReader = new QueryStringReader(),
+          qsPairs = qsReader.getAttributesAndValues();
+    let returnQueryString = '';
+
+    qsPairs.forEach(pair => {
+      if (pair.name !== '' && pair.name !== 'featureon' && pair.name !== 'featureoff')
+        returnQueryString += `${pair.name}=${pair.value}&`;
+    });
+
+    returnQueryString += featureCommandQueryString;
+    
+    return returnQueryString;
+  }
+}
+
+export interface IDisplayPanelFactory {
+  createDisplayPanel: () => DisplayPanel;
+}
+
+export class FeatureToggleDisplayPanelFactory implements IDisplayPanelFactory {
```

Comment:
This is implemented using an Abstract Factory pattern, based on the desire to be able to generate multiple types of display panels with minimal knowledge on the part of the consumer.

I'm not sure if these should be split into a separate file at this point or not.